zedd-app/src/components/AppBody.tsx

```diff
@@ -259,6 +261,41 @@ export const AppBody = observer(
                 slice.start = newStart
                 if (!isEqual(ns2, newStart)) console.warn('...')
               }}
+              onSliceStartEndChange={(slice, newStart, newEnd, oldStart, oldEnd) => {
+                const prevSlice = state.getPreviousSlice(slice)
+                const nextSlice = state.getNextSlice(slice)
+                const sliceSize = differenceInMinutes(slice.start, slice.end)
+                let distance = 0
+
+                if (prevSlice && nextSlice) {
+                  distance = differenceInMinutes(prevSlice.end as Date, nextSlice.start as Date)
+                }
+
+                slice.setInterval(newStart, newEnd)
+
+                window.onmouseup = () => {
```

Comment:
That onSliceStartEndChange is being called during drag and drop is an implementation detail of the Calendar component. This handler should not add or remove event handlers.

When it the slice doesn't fit, it shouldn't be reset on the mouseup event, it should be reset while dragging, so it can be seen immediately that it won't fit here.

This handler can be simplified: instead of accepting old start/end values, it can instead just return `false` if the passed value is not valid and let the caller handle that itself.

Reply:
Thank you for your comment.
The problem is that the slice can be pushed over several days. And if the slice is accidentally dragged over a day where everything is full, it will automatically be thrown back. This would affect usability. Currently, the user can leave the slice dragged as long as he wants and only when it is released the individual cases are checked. If the slice is thrown back immediately, the user is effectively taken away from control. What do you think about it?